Issue/remote app edit

sharanyavinod opened the pull request:
- `toAEMPath`: Added new helper for transforming remote path to AEM specific path for enabling routing.This is a stop gap solution and will be removed later.
- Issue fixes for enabling edit of remote component in AEM instance

grubyak commented on src/ModelManager.ts:
```diff
@@ -224,9 +224,7 @@ export class ModelManager {
-        const aemApiHost = this.modelClient.apiHost;
-        const isRemoteApp = PathUtils.isBrowser() && aemApiHost && (PathUtils.getCurrentURL() !== aemApiHost);
-        const currentPathname = !isRemoteApp ? PathUtils.getCurrentPathname() : '';
+        const currentPathname = !this._isRemoteApp() ? PathUtils.getCurrentPathname() : '';
```
`this._isRemoteApp() ? '' : PathUtils.getCurrentPathname();`

grubyak commented on src/ModelManager.ts:
```diff
@@ -552,6 +566,36 @@ export class ModelManager {
     }
+
+    /**
+     * Fetches parent page information of the given component path
+     * Returns object containing
+     * 1. Parent page path
+     * 2. Parent page data if already available in the store
+     * @return {object}
+     * @private
+     */
+    private _fetchParentPage(path: string) {
```
that's not how we do in typescript :)
- please declare return type (without comments if properties are self explaining), or maybe it's even declared somewhere
- no need to add `@private`

for example:
```
type PageData = {
    pageData: string;
    pagePath: string;
};

...

    /**
     * Returns model for specified path.
     */
    private _fetchParentPage(path: string): PageData {
```

grubyak replied:
if `pageData` is not a `string`, please declare dedicated type :)

grubyak commented on src/PathUtils.ts:
```diff
@@ -516,4 +516,30 @@ export class PathUtils {
+        const isLoadedInAEM = window.location.origin === aemHost;
+
+        if (isLoadedInAEM) {
+            const wcmMode = PathUtils.getMetaPropertyValue("cq:wcmmode");
+            const isEditorMode = wcmMode === 'edit';
+            const editorPrefix = isEditorMode ? '(/editor.html)?' : '';
```
```
const isAuthoring = AuthoringUtils.isStateActive(Constants.STATE_AUTHORING);
const editorPrefix = isAuthoring ? '(/editor.html)?' : '';
```

grubyak commented on src/PathUtils.ts:
```diff
@@ -516,4 +516,30 @@ export class PathUtils {
+     * @private
+     * @returns Updated url
+     */
+    public static toAEMPath(path: string, aemHost: string, rootPath: string): string {
```
`toAEMPath` does not check if `path` already is prefixed, if you wrap path more than once with this function we will end up with a broken path

grubyak replied:
`@private` and `public static toAEMPath(...)`?

grubyak commented on test/PathUtils.test.ts:
```diff
@@ -431,4 +431,29 @@ describe('PathUtils ->', () => {
         assert.equal(PathUtils.getParentNodePath('/foobar/a/xyz'), '/foobar/a');
     });
+
+    describe('toAemPath', () => {
```
should be `toAEMPath`

grubyak commented on test/PathUtils.test.ts:
```diff
@@ -431,4 +431,29 @@ describe('PathUtils ->', () => {
+
+            expect(REMOTE_APP_PAGE).toMatch(newPathPattern);
+        });
+    });
```
we are missing tests:
- when `isLoadedInAEM` is false
- when `isLoadedInAEM` is true, and wcm mode is not `edit`

grubyak reviewed (changes requested):
please relate to my comments and provide pr description

sharanyavinod commented:
> please relate to my comments and provide pr description

@grubyak Thanks for the review! I've addressed all of your comments. Please take a look again when possible.

CezCz commented on src/ModelManager.ts:
```diff
@@ -552,6 +571,34 @@ export class ModelManager {
+     * 2. Parent page data if already available in the store
+     * @return {object}
+     */
+    private _fetchParentPage(path: string): ParentPage {
```
Is it fetching in terms of doing a call?

sharanyavinod replied:
@CezCz Retrieving from the ModelStore - is the naming confusing in this context?

grubyak replied:
i am just wondering if type could be more generic than `ParentPage`? but if you are convinced it cannot be reused, i am fine with the name
